### PR TITLE
Used generator to parse/stream *.osm.pbf files for increased memory efficiency 

### LINKED ISSUE #2: `yield` (instead of appending to a `dict`) parsed data for performance increase on large files
Since some `*.osm.pbf`  files are quite large, do you think it would be possible to `yield` each row of data instead of appending it to a `dict`? So that `parse_osm_pbf()` would become a generator. This would be great for my use-case, if you want I can start a PR for this feature if you think its useful/feasible.

### PR DESCRIPTION
**Pull request by manesioz**
fixes #2 
- Added `stream_osm_pbf()` generator that returns iterator with parsed data 
- Updated `requirements.txt` accordingly 
- Updated `README.md` accordingly 

Note: This function streams layer-specific data. The reason for this was due to the fact that you may often want to stream data to a database, and each layer has a different schema, so for consistency sake I made it layer-specific.

### CODE REVIEW

File: pydriosm/read_GeoFabrik.py [hunk 1]
```diff
@@ -554,6 +556,34 @@ def parse_osm_pbf(path_to_osm_pbf, chunks_no, parsed, fmt_other_tags, fmt_single
...
+    
+    -------
+    Example
+    -------
+    
+    stream_data = stream_osm_pbf('/algeria.osm.pbf', 'lines')
+    
+    for row in stream_data: 
+        print(row)
+    '''
```
Comment (mikeqfu) [inline pydriosm/read_GeoFabrik.py]:
```python
"""
:param path_to_osm_pbf: [str] path to the *.osm.pbf file you wish to parse and stream
:param layer_name: [str] 'points', 'lines', 'multilinestrings', 'multipolygons', & 'other_relations'
:return: [generator] allowing for streaming the *.osm.pbf data of the given layer

Testing e.g.
    import pydriosm as dri
    subregion_name = 'Rutland'
    dri.download_subregion_osm_file(subregion_name, osm_file_format='.osm.pbf', update=False)
    _, path_to_osm_pbf = dri.get_default_path_to_osm_file(subregion_name, osm_file_format='.osm.pbf')

    def test_stream_osm_pbf():
        for lyr_name in ['lines', 'multilinestrings', 'points', 'multipolygons', 'other_relations']:
            stream_data = stream_osm_pbf(path_to_osm_pbf, lyr_name)
            print("First row of '{}': ".format(lyr_name))
            for row in stream_data:
                print(row)
                break


>>> test_stream_osm_pbf()
First row of 'lines': 
First row of 'multilinestrings': 
First row of 'points': 
{'type': 'Feature', 'geometry_type': 'Point', 'geometry_coordinates_0': -0.5134241, 'geometry_coordinates_1': 52.6555853, 'properties_osm_id': '488432', 'properties_name': None, 'properties_barrier': None, 'properties_highway': None, 'properties_ref': None, 'properties_address': None, 'properties_is_in': None, 'properties_place': None, 'properties_man_made': None, 'properties_other_tags': '"odbl"=>"clean"', 'id': 488432}
First row of 'multipolygons': 
First row of 'other_relations': 
"""
```

File: pydriosm/read_GeoFabrik.py [hunk 2]
```diff
@@ -554,6 +556,34 @@ def parse_osm_pbf(path_to_osm_pbf, chunks_no, parsed, fmt_other_tags, fmt_single
...
+    
+    raw_data = ogr.Open(path_to_osm_pbf)
+    layer = raw_data.GetLayer(layer_name)
+    for feature in layer: 
+        yield flatten(json.loads(feat.ExportToJson()))
```
Comment (mikeqfu) [inline pydriosm/read_GeoFabrik.py]: I would suggest you may change some variable names.

```python
raw_osm_pbf = ogr.Open(path_to_osm_pbf)
layer = raw_osm_pbf.GetLayer(layer_name)
for feat in layer:
    yield flatten_json.flatten(rapidjson.loads(feat.ExportToJson()))
```

**Comment by manesioz** [comment]
I apologize for the delay as well, I will get to this as soon as I can. You're right, it only seems to be parsing the `points` layer which is odd, I will look into that more before the other style adjustments. Thank you for your feedback!

**Comment by mikeqfu** [comment]
> I apologize for the delay as well, I will get to this as soon as I can. You're right, it only seems to be parsing the `points` layer which is odd, I will look into that more before the other style adjustments. Thank you for your feedback!

Hi @manesioz , You might have noticed that I have made some changes to the package and here is an updated version, but didn't change anything on the functions relevant to this pull request. I just wanted to know if you're still interested in working on this pull request. If so, I'd be happy to work together and help if needed. Otherwise, I might want to close this pull request and move on. Feel free to let me know what you think.

**Comment by manesioz** [comment]
Hey Mike, I am still interested I have just been busy lately so I apologize for the delay. I could not easily identify why only the `points` layer was parsed, do you have any ideas?

**Comment by mikeqfu** [comment]
> Hey Mike, I am still interested I have just been busy lately so I apologize for the delay. I could not easily identify why only the `points` layer was parsed, do you have any ideas?

I also found it very strange. Probably there was something wrong with the GDAL/OGR installation and/or settings. I couldn't figure out where exactly the problem is.

**Comment by manesioz** [comment]
Yes, possibly.. It's harder to debug GDAL/OGR since I'm not familiar with how it works under-the-hood. If you want you can close this for the time being, since I currently don't have much time to work on it. Thanks for your work, its a great package.

**Pull request closed without merging**